code/game/objects/micro_structures.dm

```diff
@@ -0,0 +1,201 @@
+/obj/structure/micro_tunnel
+	name = "mouse hole"
+	desc = "A tiny little hole... where does it go?"
+	icon = 'icons/obj/structures/micro_structures.dmi'
+	icon_state = "mouse_hole"
+
+	anchored = TRUE
+	density = FALSE
+
+	var/max_accepted_scale = 0.3
+	var/magic = FALSE	//For events and stuff, if true, this tunnel will show up in the list regardless of whether it's in valid range, of if you're in a tunnel with this var, all tunnels of the same faction will show up redardless of range
+
+/obj/structure/micro_tunnel/Initialize()
+	. = ..()
+	if(name == "mouse hole")
+		var/area/our_area = get_area(src)
+		name = "[our_area.name] mouse hole"
+	if(pixel_x || pixel_y)
+		return
+	offset_tunnel()
+
+/obj/structure/micro_tunnel/Destroy()
+	visible_message("<span class = 'warning'>\The [src] collapses!</span>")
+	for(var/mob/thing in src.contents)
+		visible_message("<span class = 'warning'>\The [thing] tumbles out!</span>")
+		thing.forceMove(get_turf(src.loc))
+
+	return ..()
+
+/obj/structure/micro_tunnel/set_dir(new_dir)
+	. = ..()
+	offset_tunnel()
+
+/obj/structure/micro_tunnel/proc/offset_tunnel()
+
+	pixel_x = 0
+	pixel_y = 0
+
+	switch(dir)
+		if(1)
+			pixel_y = 32
+		if(2)
+			pixel_y = -32
+		if(4)
+			pixel_x = 32
+		if(8)
+			pixel_x = -32
+
+/obj/structure/micro_tunnel/attack_hand(mob/user)
+	if(!isliving(user))
+		return ..()
+	if(user.loc == src)
```

Comment:
Can you actually properly attack_hand the tunnel while inside it? If so disregard. Although I'd be mildly hmmm if you can interact with anything around from inside the tunnel as well.

Reply:
You can deffo click things while you're inside them!